File: sample.cpp
#include <vector>
#include <iostream>
#include "util.hpp"

using namespace std;

double treegkr(Tree T, vector<double> creation, vector<double> destruction, vector<double> x, vector<double> y);

int main(int argc, char **argv){

  int n;
  cin >> n;

  Tree T(n);
  for(int i = 0; i < n-1; i++){
    int a, b;
    double l;
    cin >> a >> b >> l;
    a--;
    b--;
    T.add_edge(a, b, l);
  }

  vector<double> creation(n);
  vector<double> destruction(n);
  for(int i = 0; i < n; i++){
    cin >> creation[i] >> destruction[i];
  }

  vector<double> x(n);
  vector<double> y(n);
  for(int i = 0; i < n; i++){
    cin >> x[i] >> y[i];
  }
    
  double result = treegkr(T, creation, destruction, x, y);

  cout << result << endl;
  
  return 0;
}

File: sample.py
from treegkr import treegkr

def main():
    n = int(input())

    vs = []
    us = []
    cost = []
    for i in range(n-1):
        l = input().split()
        vs.append(int(l[0]) - 1)
        us.append(int(l[1]) - 1)
        cost.append(int(l[2]))

    creation = []
    destruction = []
    for i in range(n):
        c, d = map(float, input().split())
        creation.append(c)
        destruction.append(d)

    x = []
    y = []
    for i in range(n):
        a, b = map(float, input().split())
        x.append(a)
        y.append(b)

    print(treegkr(vs, us, cost, creation, destruction, x, y))
        
if __name__ == '__main__':
    main()


